errors/error-list.js

```diff
@@ -0,0 +1,22 @@
+const errorMap = {
+  signUpError: [
+    {
+      message: '信箱已經註冊過了！',
+      code: 's001',
+    },
+    {
+      message: '信箱為必填欄位！',
+      code: 's002',
+    },
+    {
+      message: '名字為必填欄位！',
+      code: 's003',
+    },
+    {
+      message: '密碼為必填欄位',
+      code: 's004',
+    },
+  ],
+};
```

Comment:
```suggestion
const signUpErrorMap = {
  duplicateEmail:  {
      message: '信箱已經註冊過了！',
      code: 's001',
    },
  EmailRequired: {
      message: '信箱為必填欄位！',
      code: 's002',
    },...
};

module.exports = signUpErrorMap

// caller
throw(new GeneralError(signUpErrorMap.duplicateEmail))
```
既然是不同的理由區分，直接用「理由」本身來區分，意思是使用時不要用 `signUpError[0]` 這種完全無法從使用方看出線索的方式當作介面。另外直接用背景（以這個為例，用SignUp 作為一個context ）來切分檔案：`errors/signUpError.js`

Reply:
ok